SETUP/upgrade/14/20200404_normalize_project_details.php

```diff
@@ -0,0 +1,53 @@
+<?php
+$relPath='../../../pinc/';
+include_once($relPath.'base.inc');
+include_once($relPath.'unicode.inc');
+
+header('Content-type: text/plain');
+
+// ------------------------------------------------------------
+
+echo "Normalizing project details...";
+
+$sql = "
+    SELECT projectid, nameofwork, authorsname, comments, extra_credits
+    FROM projects
+    ORDER BY projectid;
+";
+
+$result = mysqli_query(DPDatabase::get_connection(), $sql) or die( mysqli_error(DPDatabase::get_connection()) );
+
+$projects = array();
+while($row = mysqli_fetch_assoc($result))
+{
+    $projects[] = $row;
+}
+
+echo count($projects) . " possible projects to normalize.\n";
+
+foreach($projects as $project)
```

Comment:
To prevent loading all of the project data into memory (the `comments` field in particular is rather large) we can move the query iteration to this part and do:
```
while($project = mysqli_fetch_assoc($result))
{
```

We lose the "X possible projects.." message, but I think the trade-off is worth it.